launch/launch/substitutions/string_join_substitution.py

```diff
@@ -76,6 +96,19 @@ def delimiter(self) -> List[Substitution]:
         """Getter for delimiter."""
         return self.__delimiter
 
+    @classmethod
+    def parse(cls, data: Sequence[SomeSubstitutionsType]):
```

Comment:
Type annotation in `launch` has progressed a lot lately. Could you rebase on `rolling` and annotate the return type? I think this should do it:

```suggestion
    def parse(
        cls, data: Sequence[SomeSubstitutionsType],
    ) -> Tuple[Type['StringJoinSubstitution'], Dict[str, Any]]:
```

Reply:
Changed in 49dee17 (including imports)